src/wp-includes/class-wp-application-passwords.php

```diff
@@ -0,0 +1,801 @@
+		add_action( 'admin_post_authorize_application_password', array( __CLASS__, 'authorize_application_password' ) );
+		self::fallback_populate_username_password();
+	}
+
+	/**
+	 * Prevent caching of unauthenticated status.  See comment below.
+	 *
+	 * We don't actually care about the `wp_rest_server_class` filter, it just
+	 * happens right after the constant we do care about is defined.
+	 *
+	 * @since ?.?.0
+	 *
+	 * @access public
+	 * @static
+	 */
+	public static function wp_rest_server_class( $class ) {
+		global $current_user;
+		if ( defined( 'REST_REQUEST' )
+			&& REST_REQUEST
+			&& $current_user instanceof WP_User
+			&& 0 === $current_user->ID ) {
+			/*
+			 * For our authentication to work, we need to remove the cached lack
+			 * of a current user, so the next time it checks, we can detect that
+			 * this is a rest api request and allow our override to happen.  This
+			 * is because the constant is defined later than the first get current
+			 * user call may run.
+			 */
+			$current_user = null;
+		}
+		return $class;
+	}
+
+	/**
+	 * Handle declaration of REST API endpoints.
+	 *
+	 * @since ?.?.0
+	 *
+	 * @access public
+	 * @static
+	 */
+	public static function rest_api_init() {
+		// List existing application passwords.
+		register_rest_route( '2fa/v1', '/application-passwords/(?P<user_id>[\d]+)', array(
+			'methods' => WP_REST_Server::READABLE,
+			'callback' => __CLASS__ . '::rest_list_application_passwords',
+			'permission_callback' => __CLASS__ . '::rest_edit_user_callback',
+		) );
+
+		// Add new application passwords.
+		register_rest_route( '2fa/v1', '/application-passwords/(?P<user_id>[\d]+)/add', array(
+			'methods' => WP_REST_Server::CREATABLE,
+			'callback' => __CLASS__ . '::rest_add_application_password',
+			'permission_callback' => __CLASS__ . '::rest_edit_user_callback',
+			'args' => array(
+				'name' => array(
+					'required' => true,
+				),
+			),
+		) );
+
+		// Delete an application password.
+		register_rest_route( '2fa/v1', '/application-passwords/(?P<user_id>[\d]+)/(?P<slug>[\da-fA-F]{12})', array(
+			'methods' => WP_REST_Server::DELETABLE,
+			'callback' => __CLASS__ . '::rest_delete_application_password',
+			'permission_callback' => __CLASS__ . '::rest_edit_user_callback',
+		) );
+
+		// Delete all application passwords for a given user.
+		register_rest_route( '2fa/v1', '/application-passwords/(?P<user_id>[\d]+)', array(
+			'methods' => WP_REST_Server::DELETABLE,
+			'callback' => __CLASS__ . '::rest_delete_all_application_passwords',
+			'permission_callback' => __CLASS__ . '::rest_edit_user_callback',
+		) );
+
+		// Some hosts that run PHP in FastCGI mode won't be given the Authentication header.
+		register_rest_route( '2fa/v1', '/test-basic-authorization-header/', array(
+			'methods' => WP_REST_Server::READABLE . ', ' . WP_REST_Server::CREATABLE,
+			'callback' => __CLASS__ . '::rest_test_basic_authorization_header',
+			'permission_callback' => '__return_true',
+		) );
+	}
```

Comment:
These declarations should probably go out to `wp-includes/rest-api/endpoints/class-wp-rest-users-controller`